include/nbl/builtin/hlsl/bxdf/common.hlsl

```diff
@@ -485,14 +505,15 @@ struct SIsotropicMicrofacetCache
         NBL_CONST_REF_ARG(IsotropicInteraction) interaction,
         NBL_CONST_REF_ARG(LS) _sample)
     {
-        return createForReflection(interaction.getNdotV(), _sample.getNdotL(), _sample.getVdotL());
+        return createForReflection(interaction.getNdotV(), _sample.getNdotL(), hlsl::dot<vector3_type>(interaction.getV().getDirection(), _sample.getL().getDirection()));
     }
 
     // transmissive cases need to be checked if the path is valid before usage
     static this_t create(const bool transmitted, NBL_CONST_REF_ARG(ComputeMicrofacetNormal<scalar_type>) computeMicrofacetNormal, const scalar_type VdotL,
         NBL_CONST_REF_ARG(vector3_type) N, NBL_REF_ARG(vector3_type) H)
     {
         this_t retval;
+        retval.VdotL = VdotL;
         H = computeMicrofacetNormal.normalized(transmitted);
         retval.NdotH = hlsl::dot<vector3_type>(N, H);
```

Comment:
the variable needs to be `absNdotH`, we should adjust the getter's name accordingly, why? Because H is defined only ober the upper hemisphere

Reply:
and `H` needs to be flipped to be consistent with that (the `computeMicrofacetNormal`  doesn't care about that because it doesn't know N)

Reply:
Continued in https://github.com/Devsh-Graphics-Programming/Nabla/pull/916/files#r2310049891 and https://github.com/Devsh-Graphics-Programming/Nabla/pull/916/files#r2310051923